src/web/hooks/usePageFilter.js

```diff
@@ -0,0 +1,184 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import {useCallback, useEffect, useState} from 'react';
+
+import {useLocation, useHistory} from 'react-router-dom';
+
+import {useDispatch} from 'react-redux';
+
+import {ROWS_PER_PAGE_SETTING_ID} from 'gmp/commands/users';
+
+import Filter, {
+  DEFAULT_FALLBACK_FILTER,
+  DEFAULT_ROWS_PER_PAGE,
+  RESET_FILTER,
+} from 'gmp/models/filter';
+
+import {isDefined, hasValue} from 'gmp/utils/identity';
+
+import getPage from 'web/store/pages/selectors';
+import {pageFilter as setPageFilter} from 'web/store/pages/actions';
+import {loadUserSettingDefault} from 'web/store/usersettings/defaults/actions';
+import {getUserSettingsDefaults} from 'web/store/usersettings/defaults/selectors';
+import {loadUserSettingsDefaultFilter} from 'web/store/usersettings/defaultfilters/actions';
+import {getUserSettingsDefaultFilter} from 'web/store/usersettings/defaultfilters/selectors';
+
+import useGmp from 'web/utils/useGmp';
+
+import useShallowEqualSelector from './useShallowEqualSelector';
+
+/**
+ * Hook to get the default filter of a page from the store
+ *
+ * @param {String} pageName
+ * @returns Array of the default filter and and error if the filter could not be loaded
+ */
+const useDefaultFilter = pageName =>
+  useShallowEqualSelector(state => {
+    const defaultFilterSel = getUserSettingsDefaultFilter(state, pageName);
+    return [defaultFilterSel.getFilter(), defaultFilterSel.getError()];
+  });
+
+/**
+ * Hook to get and update the filter of a page
+ *
+ * @param {String} pageName Name of the page
+ * @param {Object} options Options object
+ * @returns Array of the applied filter, boolean indicating if the filter is
+ *          still loading, function to change the filter, function to remove the
+ *          filter and function to reset the filter
+ */
+const usePageFilter = (
+  pageName,
+  {fallbackFilter, locationQueryFilterString} = {},
+) => {
+  const gmp = useGmp();
+  const dispatch = useDispatch();
+  const location = useLocation();
+  const history = useHistory();
+
+  // only use location directly if locationQueryFilterString is undefined
+  // use null as value for not set at all
+  if (!isDefined(locationQueryFilterString)) {
+    locationQueryFilterString = location?.query?.filter;
```

Comment:
Question: Would it be possible to avoid mutating the directly `locationQueryFilterString`?
Something like this maybe.
`initialLocationQueryFilterString` passed as `arg` to the hook
```jsx
  let locationQueryFilterString = isDefined(initialLocationQueryFilterString)
    ? initialLocationQueryFilterString
    : location?.query?.filter;
```